File: imooc_music/lib_common_ui/src/main/java/net/away0x/lib_common_ui/base/BaseActivity.java
package net.away0x.lib_common_ui.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.FragmentActivity;

import net.away0x.lib_common_ui.utils.StatusBarUtil;

public class BaseActivity extends FragmentActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        StatusBarUtil.statusBarLightMode(this); // 设置 status bar 样式 (设置状态栏透明)
    }
}

File: ktmall/lib_base/src/main/java/net/away0x/lib_base/ui/activity/BaseActivity.kt
package net.away0x.lib_base.ui.activity

import android.content.Context
import android.os.Bundle
import android.view.inputmethod.InputMethodManager
import com.trello.rxlifecycle3.components.support.RxAppCompatActivity
import net.away0x.lib_base.common.AppManager

open class BaseActivity: RxAppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        AppManager.instance.addActivity(this)
    }

    override fun onDestroy() {
        super.onDestroy()
        AppManager.instance.finishActivity(this)
    }

    fun hideInputBoard() {
        // 收起键盘
        val inputMehtodManager = this.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        inputMehtodManager.hideSoftInputFromWindow(this.window.decorView.windowToken, InputMethodManager.HIDE_NOT_ALWAYS)
    }

}